Sources/AWSSDKSwiftCore/AWSClient+HAL.swift

```diff
@@ -0,0 +1,39 @@
+import HypertextApplicationLanguage
+import Foundation
+
+extension AWSClient {
+    /// process hal+json date. Extract properties from HAL
+    func hypertextApplicationLanguageProcess(response: AWSResponse, members: [AWSShapeMember]) throws -> AWSResponse {
+        switch response.body {
```

Comment:
```
guard case let .json(let data) = response.body,
  let contentType = response.headers["Content-Type"] as? String,
  content.contains("hal+json")
  else 
{
  return response
}
```

with this we can save two steps of indentation in the main code.

Reply:
done